File: Darts/src/code/screen/HumanCreationDialog.java
package code.screen;

import java.awt.BorderLayout;

import javax.swing.JPanel;

import code.db.PlayerEntity;

import java.awt.FlowLayout;

public class HumanCreationDialog extends AbstractPlayerCreationDialog
{
	public HumanCreationDialog()
	{
		super();
		
		setTitle("New Player");
		setSize(350, 225);
		setResizable(false);
		setModal(true);
		FlowLayout flowLayout = (FlowLayout) panel.getLayout();
		flowLayout.setHgap(20);
		
		getContentPane().add(panel, BorderLayout.CENTER);
		
		panel.add(avatar);
		panel.add(textFieldName);
		textFieldName.setColumns(10);
	}
	
	private final JPanel panel = new JPanel();
	
	public void init()
	{
		createdPlayer = false;
		textFieldName.setText("");
		avatar.init(null, false);
	}
	
	@Override
	protected void savePlayer()
	{
		String name = textFieldName.getText();
		long avatarId = avatar.getAvatarId();
		
		PlayerEntity.factoryAndSaveHuman(name, avatarId);
		
		createdPlayer = true;
		
		//Now dispose the window
		dispose();
	}
}

File: Darts/src/main/kotlin/dartzee/screen/HumanCreationDialog.kt
package dartzee.screen

import dartzee.db.PlayerEntity
import java.awt.BorderLayout
import java.awt.FlowLayout
import javax.swing.JPanel

class HumanCreationDialog : AbstractPlayerCreationDialog()
{
    private val panel = JPanel()

    init
    {
        title = "New Player"
        setSize(350, 225)
        isResizable = false
        isModal = true
        val flowLayout = panel.layout as FlowLayout
        flowLayout.hgap = 20

        contentPane.add(panel, BorderLayout.CENTER)

        panel.add(avatar)
        panel.add(textFieldName)
        textFieldName.columns = 10
    }

    fun init()
    {
        createdPlayer = false
        textFieldName.text = ""
        avatar.init(null, false)
    }

    override fun savePlayer()
    {
        val name = textFieldName.text
        val avatarId = avatar.avatarId

        PlayerEntity.factoryAndSaveHuman(name, avatarId)

        createdPlayer = true

        //Now dispose the window
        dispose()
    }
}
